src/CommunityToolkit/Xamarin.CommunityToolkit/ObjectModel/Extensions/INotifyPropertyChangedEx.shared.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.ComponentModel;
+
+namespace Xamarin.CommunityToolkit.ObjectModel.Extensions
+{
+	public static class INotifyPropertyChangedEx
+	{
+		public static void WeakSubscribe<T>(this INotifyPropertyChanged target, T subscriber, Action<T, object, EventArgs> action)
+		{
+			var weakSubscriber = new WeakReference(subscriber, false);
+			target.PropertyChanged += handler;
+
+			void handler(object sender, PropertyChangedEventArgs e)
+			{
+				var s = (T)weakSubscriber.Target;
+				if (s is null)
+				{
+					target.PropertyChanged -= handler;
+					return;
+				}
+
+				action.Invoke(s, sender, e);
```

Comment:
Just to be safe... Or we can validate all args on this method.
```suggestion
				action?.Invoke(s, sender, e);
```

Reply:
Hm, good point. But I would do a null check earlier. If `action` is null, there is no point in subscribing in the first place.